packages/core/src/features/st-structure.ts

```diff
@@ -0,0 +1,564 @@
+                    atRule,
+                    '.' + analyzed.name,
+                    '-st-extends',
+                    getAlias(extendedSymbol) || extendedSymbol
+                );
+            }
+            // class mapping
+            if (analyzed.mappedSelector) {
+                const firstSelectorNodes = analyzed.mappedSelector.nodes;
+                if (
+                    firstSelectorNodes[0].type !== 'pseudo_class' ||
+                    firstSelectorNodes[0].value !== 'global' ||
+                    firstSelectorNodes[0].nodes?.length !== 1
+                ) {
+                    // ToDo: support non global mapping
+                    context.diagnostics.report(diagnostics.GLOBAL_MAPPING_LIMITATION(), {
+                        node: atRule,
+                        word: stringifySelectorAst(firstSelectorNodes[0]),
+                    });
+                } else {
+                    const mappedSelectorAst = firstSelectorNodes[0].nodes[0];
+                    // analyze mapped selector
+                    analyzeRule(
+                        postcss.rule({
+                            selector: stringifySelectorAst(mappedSelectorAst),
+                            source: atRule.source,
+                        }),
+                        {
+                            isScoped: false,
+                            originalNode: atRule,
+                        }
+                    );
+                    // register global mapping to class
+                    CSSClass.extendTypedRule(
+                        context,
+                        atRule,
+                        analyzed.name,
+                        '-st-global',
+                        mappedSelectorAst.nodes
+                    );
+                }
+            }
+        } else if (analyzed.type === 'part') {
+            const parentSymbol =
+                analyzed.parentAnalyze.type === 'topLevelClass'
+                    ? CSSClass.get(context.meta, analyzed.parentAnalyze.name)
+                    : analyzedDefToPartSymbol.get(analyzed.parentAnalyze);
+
+            if (!parentSymbol) {
+                // assuming analyzing @st definitions dfs - class/part must be defined
+                return;
+            }
+            const partName = analyzed.name;
+            // check re-declare
+            if (getPart(parentSymbol, partName)) {
+                const srcWord = '::' + partName;
+                context.diagnostics.report(diagnostics.REDECLARE('pseudo-element', srcWord), {
+                    node: atRule,
+                    word: srcWord,
+                });
+                return;
+            }
+            // analyze mapped selector
+            analyzeRule(
+                postcss.rule({
+                    selector: stringifySelectorAst(analyzed.mappedSelector),
+                    source: atRule.source,
+                }),
+                {
+                    isScoped: false,
```

Comment:
should be determine from the parent chain. (global mapping)

Reply:
I believe it should just be set to `true`.

Currently the parent of a pseudo-element must be a class, that is by definition local. And in the case that a class has global mapping, by definition global mapping allow to map to global (unscoped), so there shouldn't be an error on the top level class or nested parts.